### PR TITLE
py3(django): Switch to Django 1.9

### PR DESCRIPTION
**Pull request by joshuarli**
I was meaning to revert the `LAST_VERIFIED_DJANGO_VERSION` bump in https://github.com/getsentry/sentry/pull/15323, but forgot, so here's the verification.

First, I removed the old backported migrator and changed our migrator to subclass Django 1.9's native migrator.
I made modifications to our migration template to match Django 1.9's native migration template.

Then, I modified the max length in `debug_id = models.CharField(max_length=64, db_column="uuid")` to 32.
I generated a migration with `sentry django makemigrations`, and looked at it and verified that it was generated with the new template.
I ran the migration with `sentry upgrade` to make sure it worked. The `uuid` column in table `sentry_projectdsymfile` was changed to type `character varying(32)`.

I then generated a new migration changing the max length to 100, and edited it to have `is_dangerous = True`.
I ran this migration with `MIGRATION_SKIP_DANGEROUS=1 sentry upgrade`, the migration was reported to have been faked, and indeed the max length stayed the same (`character varying(32)`).

I also generated and ran a dangerous migration with just `sentry upgrade` and it got applied, as expected.

### CODE REVIEW

**Comment by joshuarli** [comment]
This should be ready, but don't merge until vendored plugins and getsentry are compatible with 1.9 as well.

**Comment by wedamija** [comment]
> This should be ready, but don't merge until vendored plugins and getsentry are compatible with 1.9 as well.
> 
> I also cleaned up compatibility code that I found for Django < 1.9.

Also need various testing:
 - Poking around on staging in general
 - Pickled models from 1.8 work unpickle in 1.9, and vice versa.
 - Tasks that take models as params work when fired to/from 1.9 (related to previous)

**Comment by joshuarli** [comment]
Gonna need a rebase once https://github.com/getsentry/sentry/pull/15402 lands, which gives vendored plugins compat with 1.9.

**Comment by joshuarli** [comment]
Ooh, I think specifically `pytest tests/snuba/api/endpoints/test_organization_group_index.py -k test_set_private` fails on Django 1.9. Repro'd on 35d1029.

**Comment by joshuarli** [comment]
Pickling should be good. See branch `tmp/testing-django-1.8-1.9-pickled-models`. I generated pickled Events on 1.8 and 1.9, then unpickled them and compared on opposing versions.

File: tests/snuba/api/endpoints/test_organization_group_index.py [hunk 1]
```diff
@@ -1180,7 +1180,7 @@ def test_set_private(self):
         response = self.get_valid_response(
             qs_params={"id": [group1.id, group2.id]}, isPublic="false"
         )
-        assert response.data == {"isPublic": False}
+        assert response.data == {"isPublic": False, "shareId": None}
```
Comment (joshuarli) [inline tests/snuba/api/endpoints/test_organization_group_index.py]: https://github.com/getsentry/sentry/pull/15570

**Review by wedamija — APPROVED** [review]

**Pull request merged**